documentation/source/coroutines.rst

```diff
@@ -202,3 +143,90 @@ type of generator function that can be iterated with ``async for``:
 
 More details on this type of generator can be found in :pep:`525`.
 
+
+.. _yield-syntax:
+
+Generator-based coroutines
+==========================
+
+.. note:: This style is no longer recommended and support may someday be removed.
+
+Prior to Python 3.5, and the introduction of :keyword:`async` and :keyword:`await`, coroutines were implemented as wrappers around generators.
+Coroutine functions would be decorated with :class:`~cocotb.decorators.coroutine` and would use :keyword:`yield` to block on other coroutines or triggers.
+You may see existing code that uses this syntax for coroutines, but do not worry, it is compatible with async coroutines.
+
+Any object that can be used in an :keyword:`await` statement can also be used in a :keyword:`yield` statement while in a generator-based coroutine;
+including triggers like :class:`~cocotb.triggers.Timer`.
+
+.. code-block:: python3
+
+    @cocotb.coroutine
+    def clock(signal, half_period, half_period_units):
```

Comment:
Maybe call this eg myclock to make it clear even to a quick reader that this isn't our Clock class when used later